lib/versioning/VersioningRequestProcessor.js

```diff
@@ -255,19 +255,13 @@ class VersioningRequestProcessor {
         // - versioning && !versionId: create a new version
         // - !versioning && versionId: update (PUT/DELETE) an existing version
         //                             if versionId === '' update master version
-        // other cases are invalid and metadata does nothing
+        // - versioning && versionId: create a new version which has
+        //   the specified versionId (e.g. for async replication)
```

Comment:
I'm not sure I agree with this description, from what I understand the `versioning` option is not checked to create a new version that also updates the master. Isn't it more like the following?
```
         // - !versioning && versionId: update or create (PUT/DELETE) a version
         //                             if versionId === '' update master version
         //                             if versionId => master version ID, update master version
         // - versioning && versionId: same as !versioning && versionId
```
?